### PR TITLE
Add registration activity

### PR DESCRIPTION
**Pull request by ArneSchulze**
- Can't automatically merge

### CODE REVIEW

File: app/src/main/java/de/ameyering/wgplaner/wgplaner/section/registration/fragment/PickDisplayNameFragment.java [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+        if (displayName != null) {
+            inputName.setText(displayName);
+        }
+
+        btnContinue.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                String displayName = inputName.getText().toString();
+
+                if (displayName != null && !displayName.equals("")) {
```
Comment (archer96) [inline app/src/main/java/de/ameyering/wgplaner/wgplaner/section/registration/fragment/PickDisplayNameFragment.java]: Better use `!displayName.isEmpty()`

File: app/src/main/java/de/ameyering/wgplaner/wgplaner/section/registration/fragment/StateEMailFragment.java [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+                }
+            }
+        });
+
+
+        return view;
+    }
+
+    private boolean isValidEmail(String email) {
+        String regex = "^(.+)@(.+)$";
```
Comment (archer96) [inline app/src/main/java/de/ameyering/wgplaner/wgplaner/section/registration/fragment/StateEMailFragment.java]: RegEx is faulty. `test@test@test` would be valid as well.

File: app/src/main/java/de/ameyering/wgplaner/wgplaner/section/registration/fragment/UploadProfilePictureFragment.java [hunk 3]
```diff
@@ -0,0 +1,212 @@
...
+                if (resultCode == getActivity().RESULT_OK) {
+
+                    selectedImage = data.getData();
+
+                    Intent intent = new Intent("com.android.camera.action.CROP");
+                    intent.setType("image/*");
+
+                    List<ResolveInfo> activities = getActivity().getPackageManager().queryIntentActivities(intent, 0);
+
+                    if (activities.size() == 0) {
```
Comment (archer96) [inline app/src/main/java/de/ameyering/wgplaner/wgplaner/section/registration/fragment/UploadProfilePictureFragment.java]: Use `activites.isEmpty()` to better show your intent.

File: app/src/main/java/de/ameyering/wgplaner/wgplaner/customview/CircularImageView.java [hunk 4]
```diff
@@ -0,0 +1,317 @@
...
+                return true;
+
+            } else {
+                float rectangularTriangleA = Math.abs(centerX - posX);
+                float rectangularTriangleB = Math.abs(centerY - posY);
+
+                float rectangularTriangleC = (float) Math.sqrt(Math.pow(rectangularTriangleA,
+                            2) + Math.pow(rectangularTriangleB, 2));
+
+                if (rectangularTriangleC <= mDrawableRadius) {
```
[the remark this answers is not in the document]
Comment (ArneSchulze) [reply app/src/main/java/de/ameyering/wgplaner/wgplaner/customview/CircularImageView.java]: Solved

File: app/src/main/java/de/ameyering/wgplaner/wgplaner/section/registration/fragment/PickDisplayNameFragment.java (hunk 1 above)
Comment (ArneSchulze) [reply app/src/main/java/de/ameyering/wgplaner/wgplaner/section/registration/fragment/PickDisplayNameFragment.java]: Solved

File: app/src/main/java/de/ameyering/wgplaner/wgplaner/section/registration/fragment/StateEMailFragment.java (hunk 2 above)
Comment (ArneSchulze) [reply app/src/main/java/de/ameyering/wgplaner/wgplaner/section/registration/fragment/StateEMailFragment.java]: Solved

File: app/src/main/java/de/ameyering/wgplaner/wgplaner/section/registration/fragment/UploadProfilePictureFragment.java (hunk 3 above)
Comment (ArneSchulze) [reply app/src/main/java/de/ameyering/wgplaner/wgplaner/section/registration/fragment/UploadProfilePictureFragment.java]: Solved

**Pull request merged**